src/Query.php

```diff
@@ -11,10 +11,14 @@ class Query extends Container {
 	protected $name;
 	protected $variables;
 	protected $fragmentDefinitions = [];
+	protected $type;
 
-	public function __construct($name = null, $variables = []) {
-		$this->name = $name;
-		$this->variables = $variables;
+	const TYPE_QUERY = 'query';
+	const TYPE_MUTATION = 'mutation';
+
+	public function __construct($name = null, $variables = [], $type = self::TYPE_QUERY) {
+		$this->type = $type;
+		parent::__construct($name, $variables);
```

Comment:
Shouldn't call parent constructor. Very different arguments. This can't work, because name and variables aren't set anymore.

Reply:
fixed. It was copy-paste from class that extends original query from your repo.